src/expr/src/scalar/func.rs

```diff
@@ -6643,11 +6717,35 @@ impl VariadicFunc {
     /// introduces nulls even when it does not.
     pub fn introduces_nulls(&self) -> bool {
         use VariadicFunc::*;
-        // Note the negation
-        !matches!(
-            self,
-            And | Or //todo: add more
-        )
+        match self {
+            Concat
+            | PadLeading
+            | Substr
+            | ArrayCreate { .. }
+            | ListCreate { .. }
+            | RecordCreate { .. }
+            | ListSliceLinear
+            | SplitPart
+            | HmacString
+            | HmacBytes
+            | ErrorIfNull
+            | DateBinTimestamp
+            | DateBinTimestampTz
+            | RangeCreate { .. }
+            | And
+            | Or => false,
+            Coalesce
+            | Greatest
+            | Least
+            | MakeTimestamp
+            | Replace
+            | JsonbBuildArray
+            | JsonbBuildObject
+            | ArrayToString { .. }
```

Comment:
`ArrayToString` should be false, unless I'm misunderstanding something. Here are the docs: https://materialize.com/docs/sql/functions/#array_to_string

Reply:
You're right---thrown off by manual handling.